www/pages/routes-and-stops/routes-and-stops-controller.js

```diff
@@ -25,61 +38,75 @@ angular.module('pvta.controllers').controller('RoutesAndStopsController', functi
   };
 
   /*
-   * Get all the routes and stops
+   * Gets all the PVTA routes.
    */
-
-  function getRoutesAndStops () {
-    $ionicLoading.show();
+  function getRoutes () {
     // RouteForage returns a promise, resolve it.
     RouteForage.get().then(function (routes) {
       RouteForage.save(routes);
       getFavoriteRoutes(routes);
       redraw();
     });
-    /*
-    * Nested function for removing stuff we don't need
-    * from each route; this makes searching easier!
-    */
-    // Remember, StopsForage returns a Promise.
-    // Must resolve it.
+  }
+  /*
+   * Gets all the PVTA stops.
+   */
+  function getStops (position) {
     StopsForage.get().then(function (stops) {
-      stops = StopsForage.uniq(stops);
-      StopsForage.save(stops);
-      getFavoriteStops(stops);
+      $scope.stops = StopsForage.uniq(stops);
+      calculateStopDistances(position);
+      getFavoriteStops(stops, position);
       redraw();
+      $ionicLoading.hide();
     });
-
-    $ionicLoading.hide();
-
   }
   /*
-   *  Only keep the details about the
-   *  route that we care about.
+   * Helper function.
+   * Given the list of route objects, this function removes properties
+   * that we don't care about and adds whether each route is favorited.
+   *
+   * IMPORTANT: before this function is called, $scope.favoriteRoutes
+   * must already be populated!
    */
-
   function prepareRoutes (routeList) {
+    // For each route, add the custom 'Liked' property and keep only
+    // the properties we care about.  Doing this makes searching easier.
     return _.map(routeList, function (route) {
       route.Liked = _.contains(_.pluck($scope.favoriteRoutes, 'RouteId'), route.RouteId);
       return _.pick(route, 'RouteId', 'RouteAbbreviation', 'LongName', 'ShortName', 'Color', 'GoogleDescription', 'Liked');
     });
   }
 
   /*
-   *  Similar to prepareRoutes, we only
-   * keep the details about each stop that are useful
-   * to us for displaying them.  It makes searching easier.
+   * See prepareRoutes (above).
    */
-
   function prepareStops (stopList) {
     return _.map(stopList, function (stop) {
       stop.Liked = _.contains(_.pluck($scope.favoriteStops, 'StopId'), stop.StopId);
-      return _.pick(stop, 'StopId', 'Name', 'Liked', 'Description');
+      return _.pick(stop, 'StopId', 'Name', 'Liked', 'Description', 'Latitude', 'Longitude', 'Distance');
+    });
+  }
+ /*
+  * Gets the list of the user's favorite routes and
+  * passes control to prepareRoutes.
+  */
+  function getFavoriteRoutes (routes) {
```

Comment:
This code was simply moved from [here](https://github.com/umts/pvta-multiplatform/pull/272/files#diff-365f07089626b424259d271203b2ff1eL147), no need to review it.